pkg/assets/admission-registration.go

```diff
@@ -0,0 +1,65 @@
+package assets
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	admissionregistrationV1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	admclientV1 "k8s.io/client-go/kubernetes/typed/admissionregistration/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+var (
+	admScheme = runtime.NewScheme()
+	admCodecs = serializer.NewCodecFactory(admScheme)
+)
+
+type validationWebhookCfg struct {
+	Client                  *admclientV1.AdmissionregistrationV1Client
+	validationWebhookConfig *admissionregistrationV1.ValidatingWebhookConfiguration
+}
+
+func (v validationWebhookCfg) Reader(objBytes []byte, renderFunc RenderFunc, params RenderParams) {
+	var err error
+	if renderFunc != nil {
+		objBytes, err = renderFunc(objBytes, RenderParams{})
+		if err != nil {
+			panic(err)
+		}
+	}
+	obj, err := runtime.Decode(admCodecs.UniversalDecoder(admissionregistrationV1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
+	}
+	v.validationWebhookConfig = obj.(*admissionregistrationV1.ValidatingWebhookConfiguration)
+}
+
+func (v validationWebhookCfg) Applier(ctx context.Context) error {
+	_, _, err := resourceapply.ApplyValidatingWebhookConfigurationImproved(ctx, v.Client, assetsEventRecorder, v.validationWebhookConfig, nil)
+	return err
+}
+
+func admissionRegistrationClient(ctx context.Context, kubeconfigPath string) *admclientV1.AdmissionregistrationV1Client {
+	restConfig, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		klog.Fatalf("failed to create admission-registration client: %w", err)
+	}
+	return admclientV1.NewForConfigOrDie(rest.AddUserAgent(restConfig, "admission-registration"))
+}
+
+func applyAdmissionRegistration(ctx context.Context, admissionRegistrations []string, applier readerApplier, render RenderFunc, params RenderParams) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	return nil
```

Comment:
Should something happen here?

Reply:
Yeah, I still need to add the webhook deployment.  Ran out of time last week.